### PR TITLE
Add constructor of Clustering from class labels

### PR DESCRIPTION
**Pull request by jg-you**
This would allow the use of clusim as a drop-in replacement for many [sklearn.metrics](https://scikit-learn.org/stable/modules/classes.html#module-sklearn.metrics) functions.

Not a general constructor, though, since it can't handle overlap or hierarchies.

Related to this PR: it would be fairly easy to add [`@classmethod`](https://docs.python.org/3/library/functions.html#classmethod) decorators to all the `.from_X` methods to allow a slightly more straightforward initialization as:

    clu = Clustering.from_cluster_labels(labels)

instead of the current

    clu = Clustering()
    clu.from_cluster_labels(labels)

### CODE REVIEW

**Comment by yy** [comment]
Thanks!! @jg-you
